skipper.go

```diff
@@ -1357,15 +1357,17 @@ func findKubernetesDataclient(dataClients []routing.DataClient) *kubernetes.Clie
 	return kdc
 }
 
-func getRedisUpdaterFunc(namespace, name string, kdc *kubernetes.Client) func() ([]string, error) {
+func getRedisUpdaterFunc(opts *Options, kdc *kubernetes.Client) func() ([]string, error) {
 	// TODO(sszuecs): make sure kubernetes dataclient is already initialized and
 	// has polled the data once or kdc.GetEndpointAdresses should be blocking
 	// call to kubernetes API
 	return func() ([]string, error) {
-		a := kdc.GetEndpointAddresses(namespace, name)
+		a := kdc.GetEndpointAddresses(opts.KubernetesRedisServiceNamespace, opts.KubernetesRedisServiceName)
 		log.Debugf("Redis updater called and found %d redis endpoints", len(a))
+
+		port := strconv.Itoa(opts.KubernetesRedisServicePort)
 		for i := 0; i < len(a); i++ {
-			a[i] = strings.TrimPrefix(a[i], "http://")
+			a[i] = net.JoinHostPort(a[i], port)
```

Comment:
Here is an obscure bug - the code modifies cached value of `a` returned from kdc.GetEndpointAddresses

Reply:
It was not observed in production becase k8s poll interval is 3 seconds and redis endpoints update interval is 10s